receiver/iisreceiver/factory_test.go

```diff
@@ -0,0 +1,61 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package iisreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/iisreceiver"
+
+import (
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+	"go.opentelemetry.io/collector/config"
+	"go.opentelemetry.io/collector/receiver/scraperhelper"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/receiver/iisreceiver/internal/metadata"
+)
+
+func TestNewFactory(t *testing.T) {
+	testCases := []struct {
+		desc     string
+		testFunc func(*testing.T)
+	}{
+		{
+			desc: "creates a new factory with correct type",
+			testFunc: func(t *testing.T) {
+				factory := NewFactory()
+				require.EqualValues(t, typeStr, factory.Type())
+			},
+		},
+		{
+			desc: "creates a new factory with valid default config",
+			testFunc: func(t *testing.T) {
+				factory := NewFactory()
+
+				var expectedCfg config.Receiver = &Config{
+					ScraperControllerSettings: scraperhelper.ScraperControllerSettings{
+						ReceiverSettings:   config.NewReceiverSettings(config.NewComponentID(typeStr)),
+						CollectionInterval: 60 * time.Second,
+					},
+					metricSettings: metadata.DefaultMetricsSettings(),
+				}
+
+				require.Equal(t, expectedCfg, factory.CreateDefaultConfig())
+			},
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.desc, tc.testFunc)
+	}
```

Comment:
There's no point in abstracting something so simple. It's fine to group related tests, but if they have nothing in common besides the fact that they are tests, we can just do:
```go
func TestGetStatsDetails(t *testing.T) {
	t.Run("Non-200 Response", func(t *testing.T) {
		...
	})

	t.Run("Success", func(t *testing.T) {
		...
	})		
}
```

Reply:
Updated